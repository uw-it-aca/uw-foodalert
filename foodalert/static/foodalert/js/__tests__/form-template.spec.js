"use strict";
import Vue from 'vue';
import Vuex from 'vuex';

import BootstrapVue from 'bootstrap-vue'
import Vuelidate from 'vuelidate'
Vue.use(Vuelidate)
Vue.use(BootstrapVue);

import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
Vue.component('font-awesome-icon', FontAwesomeIcon);

import { library } from '@fortawesome/fontawesome-svg-core'
import {
    faUtensils,
    faChevronLeft,
    faCalendar,
    faClock,
    faMapMarkerAlt,
    faClipboardCheck,
    faBell,
} from '@fortawesome/free-solid-svg-icons'

library.add(faUtensils)
library.add(faChevronLeft)
library.add(faCalendar)
library.add(faClock)
library.add(faBell)
library.add(faMapMarkerAlt)
library.add(faClipboardCheck)

import {mount, createLocalVue} from '@vue/test-utils';
import FormController from '../components/form-controller.vue';

var localVue = createLocalVue();

localVue.use(Vuex);

describe('Form Controller', function () {
    var store;

    beforeEach(function () {
        store = new Vuex.Store({
            state: {},
        })
    });

    describe('renders correctly', function () {
        test('textarea', function () {
            var wrapper = mount(FormController, {
                localVue,
                store,
                propsData: {
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        });
    });
})
