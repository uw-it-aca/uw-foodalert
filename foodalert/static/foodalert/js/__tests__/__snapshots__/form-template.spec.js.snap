// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Controller renders correctly textarea 1`] = `
<div
  class="p-0 container-fluid"
  notificationid="0"
>
  <form
    class=""
  >
    <div
      aria-labelledby="category-heading-null"
      class="form-category my-3 container"
      id="form-category"
    >
      <!---->
       
      <div
        class="ml-2 pl-4 container"
      >
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__16__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__16"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__16__BV_label_"
            >
              
            Describe the food and event 
              <br />
            </label>
            <div
              class=""
            >
               
              <textarea
                class="form-control form-control-md"
                id="textarea-null"
                placeholder="hot indian food FIUTS weekly club meeting"
                rows="2"
                wrap="soft"
              />
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
         
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__19__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__19"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__19__BV_label_"
            >
              
            Quantity 
              <br />
            </label>
            <div
              class=""
            >
               
              <textarea
                class="form-control form-control-md"
                id="textarea-null"
                placeholder="About 8 full meals"
                rows="2"
                wrap="soft"
              />
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
         
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__22__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__22"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__22__BV_label_"
            >
              
            End Time 
              <br />
            </label>
            <div
              class=""
            >
               
              <input
                class="form-control form-control-md"
                id="time-input-null"
                placeholder="6:00 PM"
                type="time"
                value=""
              />
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
         
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__25__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__25"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__25__BV_label_"
            >
              
            Location 
              <br />
            </label>
            <div
              class=""
            >
               
              <textarea
                class="form-control form-control-md"
                id="textarea-null"
                placeholder="e.g HUB 130"
                rows="1"
                wrap="soft"
              />
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
     
    <hr />
     
    <div
      aria-labelledby="category-heading-null"
      class="form-category my-3 container"
      id="food-specifications"
    >
      <h3
        class="h5 pt-2 pb-3 pl-1"
        id="category-heading-null"
      >
        Food Specifications 
      </h3>
       
      <div
        class="ml-2 pl-4 container"
      >
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__29__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__29"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__29__BV_label_"
            >
              
            Does the food contain? 
              <br />
            </label>
            <div
              class=""
            >
               
              <div
                class=""
                id="checkbox-null"
                role="group"
                tabindex="-1"
              >
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__31"
                      name="checkbox-null"
                      true-value="milk"
                      type="checkbox"
                      value="milk"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__31"
                    >
                       milk 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__32"
                      name="checkbox-null"
                      true-value="eggs"
                      type="checkbox"
                      value="eggs"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__32"
                    >
                       eggs 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__33"
                      name="checkbox-null"
                      true-value="soy"
                      type="checkbox"
                      value="soy"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__33"
                    >
                       soy 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__34"
                      name="checkbox-null"
                      true-value="fish"
                      type="checkbox"
                      value="fish"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__34"
                    >
                       fish 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__35"
                      name="checkbox-null"
                      true-value="shellfish"
                      type="checkbox"
                      value="shellfish"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__35"
                    >
                       shellfish 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__36"
                      name="checkbox-null"
                      true-value="Peanuts"
                      type="checkbox"
                      value="Peanuts"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__36"
                    >
                       Peanuts 
                    </label>
                  </div>
                  <br />
                </div>
                <div
                  class="row mx-0"
                >
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__37"
                      name="checkbox-null"
                      true-value="wheat"
                      type="checkbox"
                      value="wheat"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__37"
                    >
                       wheat 
                    </label>
                  </div>
                  <br />
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
         
        <div
          class="px-0 py-1 container-fluid"
        >
          <div
            aria-labelledby="__BVID__39__BV_label_"
            class="labelled-textarea mb-0 b-form-group form-group"
            id="__BVID__39"
            role="group"
          >
            <label
              class="col-form-label pt-0"
              for="textarea-null"
              id="__BVID__39__BV_label_"
            >
              
            Do students need to bring containers? 
              <br />
            </label>
            <div
              class=""
            >
               
              <div
                class="container"
                id="button-null"
              >
                <button
                  class="btn btn-outline-success"
                  type="button"
                >
                  Yes
                </button>
                 
                <button
                  class="btn btn-outline-success"
                  type="button"
                >
                  No
                </button>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      aria-labelledby="category-heading-null"
      class="form-category my-3 container"
      id="message-preview"
    >
      <h3
        class="h5 pt-2 pb-3 pl-1"
        id="category-heading-null"
      >
        Message Preview 
      </h3>
       
      <div
        class="ml-2 pl-4 container"
      >
        <p
          style="color: green;"
        >
          An event has just been posted! Here are the details...
        </p>
         
        <p>
          Food Served: 
        </p>
         
        <p>
          Location: 
        </p>
         
        <p>
          Amount Left: 
        </p>
         
        <p>
          Ends At: 
        </p>
         
        <p>
          
        </p>
         
        <p>
          
        </p>
         
        <br />
      </div>
    </div>
  </form>
   
  <hr />
   
  <div
    class="mb-4 d-flex justify-content-end container"
  >
    <a
      aria-disabled="true"
      class="float-right btn btn-primary btn-lg py-2 disabled"
      href="#"
      tabindex="-1"
      target="_self"
      type="submit"
    >
       Send Notification 
    </a>
  </div>
   
  <div>
    <div
      id="modal-null___BV_modal_outer_"
    >
      <div
        aria-hidden="true"
        class="modal fade"
        id="modal-null"
        role="dialog"
        style="display: none;"
      >
        <div
          class="modal-dialog modal-md"
        >
          <div
            aria-describedby="modal-null___BV_modal_body_"
            class="modal-content"
            role="document"
            tabindex="-1"
          >
            <!---->
            <div
              class="modal-body"
              id="modal-null___BV_modal_body_"
            >
              <div
                class="px-4"
              >
                <!---->
                 
                <!---->
                 
                <p
                  class="h5 text-center mt-4 mb-5 text-center font-weight-light"
                >
                     Thank you for sharing leftover food!
                </p>
                 
                <p
                  class="h5 text-center mb-3"
                >
                   Do you have a Food Distribution Permit? 
                </p>
                 
                <fieldset
                  class="b-form-group form-group"
                  id="__BVID__45"
                  role="group"
                >
                  <!---->
                  <div
                    class=""
                    role="group"
                  >
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                 
                <div
                  class="d-flex justify-content-center"
                >
                  <button
                    class="btn px-5 btn-secondary btn-lg"
                    type="button"
                  >
                    
            No
        
                  </button>
                   
                  <button
                    class="btn btn-primary btn-lg px-5 ml-2"
                    type="button"
                  >
                    
            Yes
        
                  </button>
                </div>
                 
                <br />
                 
                <!---->
                 
                <!---->
              </div>
                 
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
